models/Element/Service.php

```diff
@@ -765,7 +765,7 @@ public static function renewReferences(mixed $data, bool $initial = true, string
                 foreach ($properties as $name => $value) {
                     //do not renew object reference of ObjectAwareFieldInterface - as object might point to a
                     //specific version of the object and must not be reloaded with DB version of object
-                    if (($data instanceof ObjectAwareFieldInterface || $data instanceof DataObject\Localizedfield) && $name === 'object') {
+                    if ($data instanceof ObjectAwareFieldInterface && $name === 'object') {
```

Comment:
Note: The additional check was redundant because `Localizedfield` implements `ObjectAwareFieldInterface` since 11.0. This is just a cleanup.